lib/server/services/ComponentService.js

```diff
@@ -779,28 +820,29 @@ module.exports = function(wilsonConfig, wilsonFrameworkConfig, logger, CacheServ
       //Add behavior dependencies
       _.each(template.dependencies.behaviors, function (behaviorName) {
         try {
-          var curScripts = getBehaviorScripts(behaviorName)
-          behaviorScripts = behaviorScripts.concat(curScripts);
-          serviceScripts = [];
-
-          _.each(behaviorScripts, function (scriptPath) {
-            serviceDependencies = getServiceDependencies(scriptPath, 'behavior');
-
-            //logger.warn('serviceDependencies', serviceDependencies);
-            // Add the scripts of each service dependency
-            _.each(serviceDependencies, function (dependency) {
-              if (serviceLibrary[dependency]) {
-                //logger.warn(dependency);
-                // Get nested dependencies from the service library, and construct the service paths
-                var nestedDependencies = serviceLibrary[dependency].dependencies;
-                var servicePaths = _.pluck(_.pick(serviceLibrary, nestedDependencies), 'path');
-                serviceScripts = serviceScripts.concat(_.uniq(servicePaths));
-                serviceScripts.push(serviceLibrary[dependency].path);
-              }
+          var currBehaviorScripts = buildTimeCache.behaviorScripts[behaviorName] || getBehaviorScripts(behaviorName);
```

Comment:
Yeah, there are a bunch of cases like this that we should look into for this type of get/cache scenario.  I think when I first was putting this in, I was thinking I needed to make sure these build-time cached values were only relevant in this function (trying to avoid the "update" and just get the initial build case since new files could be added in these directories), but now we clear that cache in both cases and this whole method gets called in both update and build cases.  We should be safe to move caching there without concern.  

There are some other methods similar to this that we can build-time cache as well while we are at it.